src/Tvheadend.cpp

```diff
@@ -367,6 +367,39 @@ PVR_ERROR CTvheadend::GetChannels(bool radio, kodi::addon::PVRChannelsResultSet&
   return PVR_ERROR_NO_ERROR;
 }
 
+PVR_ERROR CTvheadend::GetChannelStreamProperties(const kodi::addon::PVRChannel& channel, std::vector<kodi::addon::PVRStreamProperty>& properties)
+{
+  CLockObject lock(m_mutex);
```

Comment:
Move obtaining the lock just before line `auto it = m_channels.find(channel.GetUniqueId());`.